x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/topn/TopNOperator.java

```diff
@@ -71,12 +71,15 @@ static final class Row implements Accountable, Releasable {
          */
         final BreakingBytesRefBuilder values;
 
-        Row(CircuitBreaker breaker, List<SortOrder> sortOrders) {
+        Row(CircuitBreaker breaker, List<SortOrder> sortOrders, int preAllocatedValueSize) {
             boolean success = false;
             try {
                 keys = new BreakingBytesRefBuilder(breaker, "topn");
                 values = new BreakingBytesRefBuilder(breaker, "topn");
```

Comment:
I'd make the constructor take a capacity here. There's a constructor for `BytesRef` that takes a capacity too. Honestly there are only 7 calls to this so I'd just add it to the existing ctor.

Reply:
++